testing/mocker/create_common.py

```diff
@@ -0,0 +1,150 @@
+import pprint
+from abc import abstractmethod
+
+import multiprocessing
+from typing import List
+from loguru import logger
+
+
+class CreateSubject(multiprocessing.Process):
+    """
+    This is s the abstract class that highlights the main way to interact with subject related creation process
+    """
+
+    def __init__(
+        self,
+        process_id,
+        index_subject,
+        path_to_mock_data,
+        list_sessions,
+        list_anatomy_modalities,
+    ):
+        """
+        Constructor that focus on create all sessions/modalities for a given subject.
+        :param process_id:
+        :param index_subject:
+        :param path_to_mock_data:
+        :param list_sessions:
+        :param list_anatomy_modalities:
+        """
+        super(CreateSubject, self).__init__()
+        self.id = process_id
+        self.index_subject = index_subject
+        self.path_to_mock_data = path_to_mock_data
+        self.list_sessions = list_sessions
+        self.list_bids_details = list_anatomy_modalities
+        self.list_bids_keywords = ["acq", "desc", "run"]
+
+    @staticmethod
+    @abstractmethod
+    def default_constructor(path_to_mock_data: str):
+        raise NotImplementedError
+
+    @abstractmethod
+    def run(self):
+        """
+        Main entry point of the multiprocess.
+        :return:
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def create_one_json_nii_pair(self, session: str, bids_details: str):
+        raise NotImplementedError
+
+    @abstractmethod
+    def generate_a_json_file(
+        self,
+        file_stem: str,
+        session: str,
+        contrast: str,
+        modality_category: str = "anat",
```

Comment:
```suggestion
        data_type: str = "anat",
```
Again with the BIDS modality ambiguity... so what is called `modality_category` here is called `data_type` in BIDS.

Now I know that data type have multiple definitions... but I would still recommend to use the BIDS term. If it's too confusing, I would be in favor of calling here `bids_data_type` explicitely.